tests/test_backend.py

```diff
@@ -288,7 +288,7 @@ def test_set_add(self):
 
         res = cache.get("add_key")
         self.assertEqual(res, "Initial value")
-
+        cache.delete("other_key")
```

Comment:
Why you added this?

Reply:
In case the clear cache in the setUp fails.

Reply:
I see, I think that `setUp` is a very bad thing, exceptions should never pass silently, especially on tests, I suggest adding the `self.cache.clear()` on `tearDown()` method and remove the try catch block, to see if there are any exceptions and to save the whole suite of tests

Reply:
Also this one could be skipped, I have a plan to refactor the test suite

Reply:
Remove this, I will fix it in the refactor, if this has to stay then a lot of other cleanup has to be done